java/shims/pom.xml

```diff
@@ -95,5 +95,15 @@
         <directory>${build.dir}/shims</directory>
       </build>
     </profile>
+    <profile>
+      <id>java17</id>
+      <dependencies>
+        <dependency>
+          <groupId>org.apache.hadoop</groupId>
+          <artifactId>hadoop-client-api</artifactId>
+          <version>${hadoop.version}</version>
```

Comment:
Note that this is the only `compile` dependency. The other places are `test` dependency.